plugin/documents.py

```diff
@@ -917,6 +923,27 @@ def _update_stored_selection_async(self) -> Tuple[Optional[sublime.Region], bool
         self._stored_selection = selection
         return changed_first_region, True
 
+    def _format_on_paste(self) -> None:
+        self.purge_changes_async()
+        clipboard_text = sublime.get_clipboard()
+        sel = self.view.sel()
+        split_clipboard_text = clipboard_text.split('\n')
+        multi_cursor_paste = len(split_clipboard_text) == len(sel) > 1
+        restore_regions = []
+        # add regions to selection, in order for lsp_format_document_range to format those regions
+        for index, region in enumerate(sel):
+            restore_regions.append(region)
+            look_text = clipboard_text
+            if multi_cursor_paste:
+                look_text = split_clipboard_text[index]
+            found_region = self.view.find(look_text, region.end(), sublime.REVERSE | sublime.LITERAL)
+            if found_region:
+                sel.add(found_region)
+        self.view.run_command('lsp_format_document_range')
+        # restore selection how it was
```

Comment:
If the "restore_regions" would be renamed to "original_selection" then I think that the code would be self-documenting then.